src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -268,47 +268,72 @@ export async function runInteractiveSave(this: DashboardContainer, interactionMo
       }
     };
 
-    let customModalTitle;
-    let newTitle = currentState.title;
-
-    if (lastSavedId) {
-      const [baseTitle, baseCount] = extractTitleAndCount(currentState.title);
-      newTitle = `${baseTitle} (${baseCount + 1})`;
+    (async () => {
+      try {
+        let customModalTitle;
+        let newTitle = currentState.title;
+
+        if (lastSavedId) {
+          const [baseTitle, baseCount] = extractTitleAndCount(newTitle);
+          let copyCount = baseCount + 1;
+          newTitle = `${baseTitle} (${copyCount})`;
+
+          // increment count until we find a unique title
+          while (
+            !(await checkForDuplicateDashboardTitle({
```

Comment:
A possible efficiency improvement for this would be to search for all SO titles like `DashboardTitle*`, then increment locally with the results. In some cases this could result in _many_ network requests. That said, this seems fairly unlikely, so it isn't a blocker.

Reply:
merging... this in, as I've opened another PR that addresses this